pkg/metrics/instances.go

```diff
@@ -0,0 +1,156 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/global"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+)
+
+const metricPrefix = "jaeger_operator_instances"
```

Comment:
This whole file needs to be testable. For instance, prove with tests that the counter increases when there's a new CR and that it decreases when a CR is removed.

Reply:
I'll write tests for this.